schemas/api/iFrameHello.schema.json

```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://fdc3.finos.org/schemas/next/api/iframeHello.schema.json",
+  "title": "iframe Hello",
+  "description": "Hello message sent by a UI iframe to the Desktop Agent proxy setup by `getAgent()` to indicate it is ready to communicate and containing initial CSS to set on the iframe.",
+  "type": "object",
+  "allOf": [
+    {
+      "$ref": "#/$defs/IframeHelloBase"
+    },
+    {
+      "$ref": "iFrameMessage.schema.json"
+    }
+  ],
+  "$defs": {
+    "IframeHelloBase": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "title": "iframeHello Message Type",
+          "const": "iframeHello"
+        },
+        "payload": {
+          "title": "iframeHello Payload",
+          "type": "object",
+          "properties": {
+            "implementationDetails": {
+              "title": "Implementation Details",
+              "type": "string",
+              "description": "Details about the UI implementation in the iframe, such as vendor and version, for logging purposes."
+            },
+            "initialCSS": {
+              "title": "Initial CSS",
+              "type": "object",
+              "description": "A constrained set of CSS properties that should be set on the iframe before it is displayed. Note `position` cannot be specified and should always be set to `fixed`.",
+              "properties": {
+                "height": {"type": "string", "title": "height", "description": "The initial height of the iframe"},
+                "width": {"type": "string", "title": "width", "description": "The initial width of the iframe"},
+                "zIndex": {"type": "string", "title": "zIndex", "description": "The initial zindex to apply to the iframe"},
+                "left": {"type": "string", "title": "left", "description": "The initial left property to apply to the iframe"},
+                "top": {"type": "string", "title": "top", "description": "The initial top property to apply to the iframe"},
+                "bottom": {"type": "string", "title": "left", "description": "The initial bottom property to apply to the iframe"},
+                "right": {"type": "string", "title": "top", "description": "The initial right property to apply to the iframe"},
+                "transition": {"type": "string", "title": "transition", "description": "The transition property to apply to the iframe"},
+                "maxHeight": {"type": "string", "title": "maxHeight", "description": "The maximum height to apply to the iframe"},
+                "maxWidth": {"type": "string", "title": "maxWidth", "description": "The maximum with to apply to the iframe"}
+              },
+              "required": ["height", "width", "top", "left"]
```

Comment:
none of these should be required.

Reply:
I disagree, some minimal amount of initial CSS should be required or you're just making up defaults. I'll take these off for now but I think it a mistake.

Reply:
Done in 7ae1421629af507d10f33304fb14c0321ac9dad2
But a poor choice in my opinion

Reply:
the default is not to set it though

Reply:
To clarify, by 'you' I mean the `getAgent()` implementation that injects the iframes. If it doesn't receive details of the size and position that the implementation is expecting then surely it has to provide its own defaults for those. We don't yet have anything in the proposal for those and they would need standardizing, rather than simply being defined in the code.

Whereas if the implementation has to provide its own minimal defaults, then we use those. We can require `width` and `height` easily enough, and can define messages that require one each of `top`/`bottom` and `left`/`right`, although quicktype will make all 4 optional when it gerates typescript from the schema... but the docs can tell what is required.

I hope that clarifies where I'm coming from. All should be optional in the restyle message, but I question not taking defaults from the implementations and where the alternative defaults will be defined.

Reply:
ok I see what you're saying now.

So, when the iframe is created (and ahead of the `IframeHello` message arriving), `get-agent` needs to set a style on it otherwise it'll appear on the screen and Look Ugly.  

I set the defaults to this:


```
export const INITIAL_CONTAINER_CSS = {
    width: "0",
    height: "0",
    position: "fixed"
}
```

I think this is what you want to add to the standard.  However, we can't just make these required in iFrameHello, because they a) need to have been set before iFrameHello happens and b) the iframe might want to set top, left, bottom and right instead.

Reply:
We're talking about the visible iframes (channels selector would be visible and the resolver becomes visible) . You can and should totally set the defaults you are using on creation to make them hidden initially, but then real dimensions need setting to make them visible and usable. Personally, I think those should come from the implementation, and the first chance we have to get those from it are on `iframeHello`... and I think we *need* the implementation to give us both a real size and a position of some sort.

`height`/`width` could be easy to require, `top` or `bottom`, `left` or `right` is perhaps harder. I can define requirements in JSON schema and document what an implementation has to provide, but quicktype will make them all optional, rather than a creating a union of this or that - we'd just have to live with that. And yes I'm aware you could decide to do `top`/`bottom`/`left`/`right` without `width`/`height`, but most people tend to be fine with providing `width`/`height`, particularly for somethign that may be moved around...

Alternatively, it all stays optional in the scehma but with a stern comment about providing enough to correctly position/style it.

Reply:
@robmoffat are you remembering to remove the width and height of 0 if an implementation doesn't specify it and just gives top/left/bottom/right? I'm not sure if the HTML standard specifies an order of precedence we can rely on...

Reply:
Otherwise added some small detail in ef5fc762d221764234c5055037123df7d2101576

Reply:
> [robmoffat](https://github.com/robmoffat) are you remembering to remove the width and height of 0 if an implementation doesn't specify it and just gives top/left/bottom/right? I'm not sure if the HTML standard specifies an order of precedence we can rely on...

Yes, they get cleared if not set in the JSON message.